gammapy/catalog/lhaaso.py

```diff
@@ -0,0 +1,192 @@
+from gammapy.utils.gauss import Gauss2DPDF
+from gammapy.utils.scripts import make_path
+from .core import SourceCatalog, SourceCatalogObject
+
+__all__ = [
+    "SourceCatalogObject1LHAASO",
+    "SourceCatalog1LHAASO",
+]
+
+
+class SourceCatalogObject1LHAASO(SourceCatalogObject):
+    """One source from the 1LHAASO catalog.
+
+    Catalog is represented by `~gammapy.catalog.SourceCatalog1LHAASO`.
+    """
+
+    _source_name_key = "Source_Name"
+
+    def _parse(self, name, which):
+        if which in self.data["Model_a"]:
+            tag = ""
+        elif which in self.data["Model_b"]:
+            tag = "_b"
+        else:
+            raise ValueError("Invalid model component name")
+        is_ul = False
+        value = u.Quantity(self.data[f"{name}{tag}"])
+        if (
+            np.isnan(value) or value == 0 * value.unit
+        ) and f"{name}_ul{tag}" in self.data:
+            value = self.data[f"{name}_ul{tag}"]
+            is_ul = True
+        return value, is_ul
+
+    def _get(self, name, which):
+        value, _ = self._parse(name, which)
+        return value
+
+    def spectral_model(self, which):
+        """Spectral model (`~gammapy.modeling.models.PowerLawSpectralModel`).
+
+        * ``which="KM2A"`` - Sky model for KM2A analysis only.
+        * ``which="WCDA"`` - Sky model for WCDA analysis only.
+
+        Entries not repeated for the second analysis are taken from the first one.
+        """
+
+        pars = {
+            "reference": self._get("E0", which),
+            "amplitude": self._get("N0", which),
+            "index": self._get("gamma", which),
+        }
+
+        errs = {
+            "amplitude": self._get("N0_err", which),
+            "index": self._get("gamma_err", which),
+        }
+
+        model = Model.create("PowerLawSpectralModel", "spectral", **pars)
+
+        for name, value in errs.items():
+            model.parameters[name].error = value
+
+        return model
+
+    def spatial_model(self, which):
+        """Spatial model (`~gammapy.modeling.models.SpatialModel`).
+
+        * ``which="KM2A"`` - Sky model for KM2A analysis only.
+        * ``which="WCDA"`` - Sky model for WCDA analysis only.
+
+        Entries not repeated for the second analysis are taken from the first one.
+        """
+
+        lat_0 = self._get("DECJ2000", which)
+        pars = {"lon_0": self._get("RAJ2000", which), "lat_0": lat_0, "frame": "fk5"}
+
+        pos_err = self._get("pos_err", which)
+        scale_r95 = Gauss2DPDF().containment_radius(0.95)
+
+        errs = {
+            "lat_0": pos_err / scale_r95,
+            "lon_0": pos_err / scale_r95 / np.cos(lat_0),
+        }
+
+        r39, is_ul = self._parse("r39", which)
+        if not is_ul:
+            pars["sigma"] = r39
+            model = Model.create("GaussianSpatialModel", "spatial", **pars)
+        else:
+            model = Model.create("PointSpatialModel", "spatial", **pars)
```

Comment:
Is it better to use a `PointSpatialModel` or a `Gaussian` one with very small `sigma`?

Reply:
I would say a `PointSpatialModel ` so we don't have to decide about the sigma to use